src/Storages/StorageMaterializedView.cpp

```diff
@@ -152,19 +158,28 @@ void StorageMaterializedView::read(
     const size_t max_block_size,
     const size_t num_streams)
 {
+    auto context = getInMemoryMetadataPtr()->getDefinerContext(local_context);
```

Comment:
Please add more comments to the code about why you're using this or that context. We talked about this, but our talk is not going to be in the code unless you write it ;)